spec/lib/kafka/avro_producer_payload_spec.rb

```diff
@@ -24,127 +21,182 @@
     }
   end
   let(:invalid_payload_format) { 'invalid' }
+  let(:topic3_payload_value) do
+    <<~PAYLOAD
+      \x00\x00\x00\x00\x05\x00
+      \n12345\x00\x02\x12ICN123456
+      \x12VASI98765\x00\x00\x00\x00\x00\x00
+      (2024-03-04T12:00:00Z\x00
+    PAYLOAD
+  end
 
   describe '#validate_payload!' do
```

Comment:
Could you rename this describe block to not directly reference the private method `#validate_payload!`? I see that you're no longer testing that method but the public `.produce` method instead, which is great. You can just call the describe block something like "validate payload" instead. 🙂